engine/src/sol/rpc.rs

```diff
@@ -118,11 +118,18 @@ async fn call_rpc_raw(
 
 	let mut json = response.json::<serde_json::Value>().await.map_err(Error::Transport)?;
 
-	if json["error"].is_object() {
-		return Err(Error::Rpc(serde_json::from_value(json["error"].clone()).map_err(Error::Json)?));
+	if json.is_object() {
+		if json["error"].is_object() {
+			return Err(Error::Rpc(
+				serde_json::from_value(json["error"].clone()).map_err(Error::Json)?,
+			));
+		}
+
+		Ok(json["result"].take())
+	} else {
+		warn!("The rpc response returned for {method:?} with params: {params:?} was not a valid json object: {json:?}");
+		Err(Error::Rpc(serde_json::from_value(json.clone()).map_err(Error::Json)?))
 	}
-
-	Ok(json["result"].take())
```

Comment:
I was wondering why it was accessing the data mutably (because immutable access doesn't panic), and now I see it must be here, since we `take` the value.

In the other file, we use `to_owned()` which clones the value.